web/client/plugins/queryPanelWithMap/SpatialFilter.jsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2023, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {connect} from "react-redux";
+import {getMapConfigSelector} from "../../selectors/widgets";
+import {bindActionCreators} from "redux";
+import {
+    changeDwithinValue,
+    changeSpatialFilterValue,
+    expandSpatialFilterPanel,
+    removeSpatialSelection,
+    selectSpatialMethod,
+    selectSpatialOperation,
+    selectViewportSpatialMethod, showSpatialSelectionDetails,
+    zoneChange, zoneGetValues, zoneSearch
+} from "../../actions/queryform";
+import {changeDrawingStatus} from "../../actions/draw";
+import SpatialFilterComponent from "../../components/data/query/SpatialFilter";
+
+const SpatialFilter = connect((state) => {
+    return {
+        useMapProjection: state.queryform.useMapProjection,
+        spatialField: state.queryform.spatialField,
+        showDetailsPanel: state.queryform.showDetailsPanel,
+        spatialPanelExpanded: state.queryform.spatialPanelExpanded,
+        zoom: (getMapConfigSelector(state) || {}).zoom,
+        projection: (getMapConfigSelector(state) || {}).projection
```

Comment:
The difference between this component and the original one stays only in these 2 selectors connection.
I think we should try to avoid the duplication introducing 2 plugins, but only 1 with some config/auto-config options.

Reply:
what do you mean exactly?

Reply:
In general I think you can work on a single plugin. 
I didn't think how to switch between using the embedded map or the map plugin. 
One solution can be via a configuration flag. Anyway if it is able to find out if the map plugin is present, it could be a smarter solution.